spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java

```diff
@@ -93,7 +95,16 @@ public ModuleDeploymentId deploy(ModuleDeploymentRequest request) {
 
 
 		try {
-			Path workDir = Files.createDirectory(Paths.get(logPathRoot.toFile().getAbsolutePath(),
+			String groupDeploymentId = request.getDeploymentProperties().get(GROUP_DEPLOYMENT_ID);
+			if (groupDeploymentId == null) {
+				groupDeploymentId = request.getDefinition().getGroup() + "-" + System.currentTimeMillis();

```

Comment:
This makes groupDeploymentId calculation appear at 2 places. Can we assume it is never null, with the "only" caveat that some unit tests may have to explicitly pass one?